src/blueprint/from_config.rs

```diff
@@ -358,6 +361,24 @@ fn update_modify(
     None => Valid::Ok(Some(b_field)),
   }
 }
+fn update_const_field(field: &config::Field, mut b_field: FieldDefinition, config: &Config) -> Valid<FieldDefinition> {
+  match field.const_field.as_ref() {
+    Some(const_field) => {
+      let data = &const_field.data;
+      if let Ok(json) = ConstValue::from_json(data.to_owned()) {
+        if to_json_schema_for_field(field, config).validate(&json).is_ok() {
+          b_field.resolver = Some(Literal(data.to_owned()));
+          Valid::Ok(b_field)
+        } else {
+          Valid::fail(format!("const data doesn't conform with the type `{}`.", field.type_of))
```

Comment:
error message should show what part of the data don't confirm to the schema.

Reply:
`validate` actually returns a validation error, which is very detailed. I think we can just return those errors directly.

Reply:
Covered this remarks in new commit.